protos/flyteidl/plugins/ray.proto

```diff
@@ -19,13 +21,19 @@ message RayCluster {
   HeadGroupSpec head_group_spec = 1;
   // WorkerGroupSpecs are the specs for the worker pods
   repeated WorkerGroupSpec worker_group_spec = 2;
+  // Namespace used to create the ray cluster
+  string namespace = 3;
```

Comment:
To make it more generic, could we add it to the task metadata [here](https://github.com/flyteorg/flyteidl/blob/0dabef70811b92d63ac4abaff44eb0c96162b3c7/protos/flyteidl/core/tasks.proto#L65)